greedy_methods/job_sequence.py

```diff
@@ -0,0 +1,60 @@
+"""
+Given a list of tasks, each with a deadline and reward,
+The maximum number of tasks that can be scheduled to maximize reward.
+We can only complete one task at a time, and each task takes 1 unit
+of time to complete. Once a task has passed its deadline, it can no
+longer be scheduled.
+
+Example :
+tasks_info = [(4, 20), (1, 10), (1, 40), (1, 30)]
+max_tasks will return (2, [2, 0]) -
+which is by scheduling the tasks with rewards 40, 20
+
+This problem can be solved using the concept of "GREEDY ALGORITHM".
+Time Complexity - O(n log n)
+
+We iterate over the tasks array once, sorting it in descending order of reward.
+Then we iterate over the sorted tasks array once more, scheduling each
+task if its deadline is greater than the current time.The greedy choice at
+each point is to either schedule the current task if its deadline is greater
+than the current time, or skip it otherwise.
+"""
+
+from typing import Any
+
+
+class Task:
+    def __init__(self, task_id: Any, deadline: int, reward: int) -> None:
+        self.task_id = task_id
+        self.deadline = deadline
+        self.reward = reward
+
+
+def max_tasks(tasks_info: list[tuple[int]]) -> int:
```

Comment:
```suggestion
def max_tasks(tasks_info: list[tuple[int, int]]) -> int:
```
greedy_methods/job_sequence.py:40: error: Need more than 1 value to unpack (2 expected)  [misc]
greedy_methods/job_sequence.py:40: error: Cannot determine type of "d"  [has-type]
greedy_methods/job_sequence.py:40: error: Cannot determine type of "p"  [has-type]
greedy_methods/job_sequence.py:52: error: Incompatible return value type (got "tuple[int, list[Any]]", expected "int")  [return-value]
greedy_methods/job_sequence.py:60: error: List item 0 has incompatible type "tuple[int, int]"; expected "tuple[int]"  [list-item]
greedy_methods/job_sequence.py:60: error: List item 1 has incompatible type "tuple[int, int]"; expected "tuple[int]"  [list-item]
greedy_methods/job_sequence.py:60: error: List item 2 has incompatible type "tuple[int, int]"; expected "tuple[int]"  [list-item]
greedy_methods/job_sequence.py:60: error: List item 3 has incompatible type "tuple[int, int]"; expected "tuple[int]"  [list-item]